enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,327 @@
+│   │   ├── etcdoperator-controller-manager-metrics-service_v1_service.yaml
+│   │   ├── etcdoperator-metrics-reader_rbac.authorization.k8s.io_v1beta1_clusterrole.yaml
+│   │   ├── etcdbackup.crd.yaml
+│   │   ├── etcdcluster.crd.yaml
+│   │   └── etcdrestore.crd.yaml
+│   └── metadata
+│       └── annotations.yaml
+└── bundle-0.9.2.Dockerfile
+```
+
+#### Complex Example
+
+```
+operator-sdk migrate bundle manifests --bundle-dir=my-bundle --overwrite
+```
+
+This will generate the bundles in a `my-bundle` directory. If there was an
+existing `my-bundle` the `--overwrite` will delete that directory before writing
+the new bundles.
+
+```
+my-bundle
+├── my-bundle-0.6.1
+│   ├── manifests
+│   │   ├── etcdoperator.clusterserviceversion.yaml
+│   │   ├── etcdoperator-controller-manager-metrics-service_v1_service.yaml
+│   │   ├── etcdoperator-metrics-reader_rbac.authorization.k8s.io_v1beta1_clusterrole.yaml
+│   │   └── etcdcluster.crd.yaml
+│   └── metadata
+│       └── annotations.yaml
+├── my-bundle-0.6.1.Dockerfile
+├── my-bundle-0.9.1
+│   ├── manifests
+│   │   ├── etcdoperator.clusterserviceversion.yaml
+│   │   ├── etcdoperator-controller-manager-metrics-service_v1_service.yaml
+│   │   ├── etcdoperator-metrics-reader_rbac.authorization.k8s.io_v1beta1_clusterrole.yaml
+│   │   ├── etcdbackup.crd.yaml
+│   │   ├── etcdcluster.crd.yaml
+│   │   └── etcdrestore.crd.yaml
+│   └── metadata
+│       └── annotations.yaml
+├── my-bundle-0.9.1.Dockerfile
+├── my-bundle-0.9.2
+│   ├── manifests
+│   │   ├── etcdoperator.clusterserviceversion.yaml
+│   │   ├── etcdoperator-controller-manager-metrics-service_v1_service.yaml
+│   │   ├── etcdoperator-metrics-reader_rbac.authorization.k8s.io_v1beta1_clusterrole.yaml
+│   │   ├── etcdbackup.crd.yaml
+│   │   ├── etcdcluster.crd.yaml
+│   │   └── etcdrestore.crd.yaml
+│   └── metadata
+│       └── annotations.yaml
+└── my-bundle-0.9.2.Dockerfile
+```
+
+#### Prototype Links
+
+- migrate-prototype: to see what `GenerateFunc` can do:
+  https://github.com/jmrodri/migrate-prototype/blob/master/main.go
+
+### Risks and Mitigations
+If we can not change the behavior of `GenerateFunc` to write the
+`bundle.Dockerfile` in a particular location this could make this process
+difficult.
+
+## Design Details
+
+### Test Plan
+
+- add unit tests for migration code
+- add e2e tests to migrate packagemanifests to bundles
+- manual testing of `migrate bundle` will be done as well
+
+### Graduation Criteria
+
+The `migrate bundle` command will come in as non-alpha and be ready to
+use once merged.
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+20210210 - Add prototype and more explicit details.
+20210209 - Generate 2 bundles
+20210208 - Initial proposal to migrate bundle
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+There are 2 alternatives that I can come up with:
+
+1. Use a different command name `convert packagemanifest` which would take the
```

Comment:
How about `bundle migrate`, since a `bundle` subcommand already exists? Or do you think that would be confusing?

Reply:
From a slack conversation, we settled on `pkgman-to-bundle` to be explicit about what it actually will do.